File: ChatGPT/Direct_Attempt/python/output/output_14a.py
# ChatGPT-generated solution will go here.
def main():
    width, height = 101, 103
    time = 100
    robots = []

    with open('input.txt') as f:
        for line in f:
            if not line.strip():
                continue
            parts = line.strip().split()
            px, py = map(int, parts[0][2:].split(','))
            vx, vy = map(int, parts[1][2:].split(','))
            robots.append((px, py, vx, vy))

    positions = {}
    for px, py, vx, vy in robots:
        x = (px + vx * time) % width
        y = (py + vy * time) % height
        positions[(x, y)] = positions.get((x, y), 0) + 1

    mid_x = width // 2
    mid_y = height // 2
    quadrants = [0, 0, 0, 0]  # Q1, Q2, Q3, Q4

    for (x, y), count in positions.items():
        if x == mid_x or y == mid_y:
            continue
        if x < mid_x and y < mid_y:
            quadrants[0] += count  # top-left
        elif x >= mid_x and y < mid_y:
            quadrants[1] += count  # top-right
        elif x < mid_x and y >= mid_y:
            quadrants[2] += count  # bottom-left
        else:
            quadrants[3] += count  # bottom-right

    safety_factor = 1
    for q in quadrants:
        safety_factor *= q

    print(safety_factor)

if __name__ == '__main__':
    main()

File: ChatGPT/Direct_Attempt/clojure/output/output_14a.clj
;; ChatGPT-generated solution will go here.
(ns output_14a
  (:require [clojure.string :as str]))

(def width 101)
(def height 103)
(def time 100)

(defn parse-line [line]
  (let [[_ px py vx vy]
        (re-matches #"p=(-?\d+),(-?\d+)\s+v=(-?\d+),(-?\d+)" line)]
    [(Long/parseLong px)
     (Long/parseLong py)
     (Long/parseLong vx)
     (Long/parseLong vy)]))

(defn move-robot [[px py vx vy]]
  (let [x (mod (+ px (* time vx)) width)
        y (mod (+ py (* time vy)) height)]
    [x y]))

(defn part [lines]
  (let [robots (map parse-line lines)
        positions (map move-robot robots)
        mx (quot width 2)
        my (quot height 2)
        quadrants (frequencies
                    (for [[x y] positions
                          :when (not (or (= x mx) (= y my)))]
                      (cond
                        (and (< x mx) (< y my)) :tl
                        (and (> x mx) (< y my)) :tr
                        (and (< x mx) (> y my)) :bl
                        (and (> x mx) (> y my)) :br)))
        counts (map #(get quadrants % 0) [:tl :tr :bl :br])]
    (reduce * counts)))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
